### ISSUE TITLE
Indefinitely receiving Insufficient Space Error 

### ISSUE THREAD

**Issue by 4kills** [ebiggers/libdeflate#279]
Hey,

thank you for this library, first of all. 

I've run into the following problem:  
Upon inputting this array:
```c
char in[] = { 0x78, 0x9c, 0x7d, 0x90, 0x31, 0x6f, 0x83, 0x30, 0x10, 0x85, 0xf7, 0xfc, 0xa, 0xc4, 0x5c, 0xac, 0x3b, 0x1b, 0x63, 0xcc, 0x6, 0x85, 0x54, 0x2c, 0x55, 0xa4, 0x26, 0x4b, 0x17, 0x84, 0x82, 0x5b, 0xd1, 0x2, 0x46, 0x86, 0x28, 0x48, 0x51, 0xfe, 0x7b, 0x81, 0x4, 0xc4, 0x54, 0xcb, 0x8b, 0xdf, 0xf7, 0xde, 0x9d, 0xef, 0x6e, 0x3b, 0x6b, 0x3c, 0xf6, 0xdb, 0x29, 0x8d, 0xed, 0xc0, 0xb2, 0x6f, 0x2, 0x38, 0x40, 0x14, 0x85, 0x8e, 0xfb, 0x2a, 0xf7, 0x8e, 0xcb, 0x15, 0x73, 0xc2, 0x24, 0xe2, 0x4e, 0x92, 0x30, 0x1e, 0x49, 0x19, 0xd1, 0x58, 0xc0, 0xdd, 0x7e, 0x79, 0xa4, 0xa, 0x5d, 0xe7, 0x65, 0xf3, 0x9e, 0xd7, 0x6a, 0xca, 0x2e, 0x6a, 0x36, 0x4c, 0xaf, 0x5a, 0x17, 0xca, 0xcd, 0xbe, 0x2a, 0x7d, 0xcd, 0x8c, 0x6a, 0xb5, 0xe9, 0xd7, 0x90, 0xea, 0xfa, 0xb4, 0x9d, 0x2c, 0xe8, 0x7a, 0xc4, 0xf3, 0x8, 0x72, 0x4a, 0x5c, 0x5c, 0x70, 0x55, 0x36, 0xbf, 0xe1, 0xb9, 0x3f, 0x96, 0x8f, 0xa2, 0x14, 0x28, 0x75, 0x70, 0xbc, 0xfc, 0x88, 0x2c, 0x60, 0x34, 0xa0, 0xfc, 0x73, 0x6b, 0xfd, 0xe8, 0x73, 0xf3, 0x8f, 0x99, 0xae, 0xe6, 0x66, 0xfc, 0xcf, 0xb3, 0xad, 0xf, 0x4, 0x3d, 0x46, 0x4, 0x12, 0x29, 0x17, 0xdc, 0x1a, 0xbd, 0xcc, 0x61, 0x78, 0xde, 0xaa, 0x81, 0xa8, 0x41, 0x6d, 0x60, 0xaf, 0xcf, 0xba, 0x9a, 0xe8, 0x29, 0x3e, 0x2c, 0xf2, 0xa5, 0x53, 0x66, 0x9, 0x75, 0x3f, 0x4, 0x19, 0x48, 0x81, 0xbe, 0x10, 0x6c, 0x1d, 0xe6, 0xd2, 0xee, 0xc7, 0x5, 0x8c, 0x9c, 0x31, 0x80, 0xcd, 0xfc, 0x87, 0x69, 0x1f, 0x81, 0x45, 0x5, 0x30, 0x5c, 0x77, 0x79, 0x6d, 0x9e, 0x66, 0x4, 0x2e, 0x37, 0x2d, 0xd2, 0x62, 0x2e, 0xc0, 0x1, 0xc1, 0x43, 0xff, 0x9, 0xbe, 0xc7, 0xbe, 0xa1, 0x51, 0xf9, 0xc, 0x5, 0xf3, 0xe9, 0x16, 0xcc, 0x22, 0xf5, 0x7c, 0xc9, 0x76, 0xf7, 0x3f, 0xc7, 0x43, 0x74 };
```
into `libdeflate_deflate_decompress_ex`, I receive continous `LIBDEFLATE_INSUFFICIENT_SPACE`. I have tried with a 16GiB output buffer. 

The implementation in another language (Golang) reports `unexpected EOF` upon entering the data. I assume the data is either corrupted or has a psychic compression factor.  
What is the problem here?

**Comment by ebiggers** [comment]
This was already fixed in v1.10 by https://github.com/ebiggers/libdeflate/pull/158 (fix for issue https://github.com/ebiggers/libdeflate/issues/157).

Before reporting issues, please check whether they still exist in the latest version.

**Comment by ebiggers** [comment]
Note that this was only an issue in applications that decompress using an unknown output size with no limit, which is not recommended.  Such applications are still vulnerable to zip bombs with up to 1032x expansion ratio (or even more if the application also supports other compression formats alongside DEFLATE, zlib, or gzip).
